probe_scraper/probe_expiry_alert.py

```diff
@@ -70,46 +74,81 @@ def find_expiring_probes(probes, target_version):
 
         if expiry_version == target_version:
             expiring_probes[name] = (
-                    details.get("notification_emails", []) + [DEFAULT_TO_EMAIL])
+                    details.get("notification_emails", []))
 
     return expiring_probes
 
 
-def send_emails_for_expiring_probes(expired_probes, expiring_probes,
-                                    current_version, dryrun=True):
-    probes_by_email_by_state = defaultdict(lambda: defaultdict(list))
+def check_bugzilla_user_exists(email, request_headers):
+    user_response = requests.get(BUGZILLA_USER_URL + "?match=" + email, headers=request_headers)
+    user_response.raise_for_status()
+    return len(user_response.json()["users"]) > 0
 
-    # Get expired probes for each email
-    for name, emails in expired_probes.items():
-        for email in emails:
-            probes_by_email_by_state[email]["expired"].append(name)
 
-    # Get expiring probes for each email
-    for name, emails in expiring_probes.items():
-        for email in emails:
-            probes_by_email_by_state[email]["expiring"].append(name)
+def search_bugs(version, request_header):
+    search_query_params = {
+        "summary": BUG_SUMMARY_TEMPLATE.format(version=version, probe=""),
+        "whiteboard": BUG_WHITEBOARD_TAG,
+    }
+    return requests.get(BUGZILLA_BUG_URL, params=search_query_params,
+                        headers=request_header).json()["bugs"]
+
+
+def create_bug(probe_name, version, emails, request_header):
+    description_notes = ("No emails associated with the probe have a "
+                         "corresponding Bugzilla account." if len(emails) == 0 else "")
+
+    create_params = {
+        "product": "Toolkit",
+        "component": "Telemetry",
+        "summary": BUG_SUMMARY_TEMPLATE.format(version=version, probe=probe_name),
+        "description": BUG_DESCRIPTION_TEMPLATE.format(
+            version=version, probe_name=probe_name, notes=description_notes),
+        "version": "unspecified",
+        "type": "task",
+        "whiteboard": BUG_WHITEBOARD_TAG,
```

Comment:
Also consider adding "See Also" relationships with the bugs in the bug_numbers field.